zjit/src/hir_type/mod.rs

```diff
@@ -585,6 +593,19 @@ mod tests {
         assert_eq!(types::Integer.inexact_ruby_class(), None);
     }
 
+    #[test]
+    fn set() {
+        assert_subtype(types::SetExact, types::Set);
+        assert_subtype(types::SetSubclass, types::Set);
+    }
+
+    #[test]
+    fn set_has_ruby_class() {
```

Comment:
This should probably use the with_vm thing because otherwise we're messing with uninitialized pointers. And even though we're not dereferencing them, it makes me uneasy